next@canary - can't find module "webpack5"

### What version of Next.js are you using?

canary

### What version of Node.js are you using?

16.11.1

### What browser are you using?

Brave

### What operating system are you using?

Windows

### How are you deploying your application?

NestJS / Express

### Describe the Bug

next@canary can't find the module "webpack5"
```
[10:53:32] Starting compilation in watch mode...

node_modules/next/dist/server/config-shared.d.ts:1:27 - error TS2307: Cannot find module 'webpack5' or its corresponding type declarations.

1 import type webpack5 from 'webpack5';
                            ~~~~~~~~~~

[10:53:37] Found 1 error. Watching for file changes.
```

### Expected Behavior

next@canary can find the module and compiles without issues

### To Reproduce

Create a express app with nextjs and typescript.
replace next with next@canary
error should occour

I tried the latest canary
`^12.0.3-canary.7`

and I get this error still
```
> tsc

node_modules/next/dist/server/config-shared.d.ts(1,15): error TS2724: '"next/dist/compiled/webpack/webpack"' has no exported member named 'webpack5'. Did you mean 'webpack'?
The command '/bin/sh -c npm run build' returned a non-zero code: 2
```

--- a/packages/next/bundles/webpack/packages/webpack.d.ts
+++ b/packages/next/bundles/webpack/packages/webpack.d.ts
@@ -2,3 +2,9 @@ export namespace webpack {
   export type Compiler = any
   export type Plugin = any
 }
+
+export namespace webpack5 {
+  export type Compiler = any
+  export type Plugin = any
+  export type Configuration = any
+}

--- a/packages/next/compiled/webpack/webpack.d.ts
+++ b/packages/next/compiled/webpack/webpack.d.ts
@@ -2,3 +2,9 @@ export namespace webpack {
   export type Compiler = any
   export type Plugin = any
 }
+
+export namespace webpack5 {
+  export type Compiler = any
+  export type Plugin = any
+  export type Configuration = any
+}

